File: scraper/models/price.go
package models

import "time"

type FuelPrice struct {
    Company   string    `json:"company"`
    FuelType  string    `json:"fuel_type"`
    Price     float64   `json:"price"`
    Unit      string    `json:"unit"`
    ScrapedAt time.Time `json:"scraped_at"`
}

File: frontend/src/types/price.ts
export interface FuelPrice {
  id: number;
  company: string;
  fuel_type: string;
  price: number;
  unit: string;
  scraped_at: string;
}

export interface Prediction {
  available: boolean;
  company: string;
  fuel_type: string;
  predicted_price: number;
  predicted_for: string;
  confidence: {
    percent: number;
    level: string;
    message: string;
    spread: number;
  };
}

export interface ApiResponse<T> {
  status: string;
  data: T;
}
